src/query.rs

```diff
@@ -23,6 +29,78 @@ pub struct FundingOutput {
     pub height: u32,
     pub output_index: usize,
     pub value: u64,
+    pub asset: Option<Asset>,
+}
+
+impl FundingOutput {
+    pub fn build(
+        txn_id: Sha256dHash,
+        height: u32,
+        output_index: usize,
+        value: u64,
+        asset: Option<Asset>,
+    ) -> Self {
+        FundingOutput {
+            txn_id,
+            height,
+            output_index,
+            value,
+            asset,
+        }
+    }
+}
+
+impl FundingOutput {
+    pub fn to_json(&self) -> Value {
+        if self.asset.is_none() {
+            json!({
+                "height": self.height,
+                "tx_pos": self.output_index,
+                "tx_hash": self.txn_id.to_hex(),
+                "value": self.value,
+            })
+        } else {
+            json!({
+                "height": self.height,
+                "tx_pos": self.output_index,
+                "tx_hash": self.txn_id.to_hex(),
+                "value": self.value,
+                "asset": self.asset.as_ref().expect("failed to read asset"),
+            })
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Serialize)]
+pub struct Asset {
+    pub asset_id: AssetId,
+    pub asset_quantity: u64,
```

Comment:
Don't you need metadata such as asset definition url.

Reply:
Yes, I need.
But I prefer to submit another PR to add metadata field after this PR because I should fix the type of metadata in openassets-rust.

https://github.com/chaintope/openassets-rust/blob/master/src/openassets/marker_output.rs#L13

It is the 'String' type, but Open Assets Protocol allows arbitrary data (including binary).